addons/account/models/account_move.py

```diff
@@ -5519,6 +5503,30 @@ def _add_cash_basis_lines_to_exchange_difference_vals(lines, exchange_diff_move_
 
         return exchange_move
 
+    def _prepare_exchange_vals_0(self, line, sequence, exchange_line_account):
+        return {
+            'name': _('Currency exchange rate difference'),
+            'debit': -line.amount_residual if line.amount_residual < 0.0 else 0.0,
+            'credit': line.amount_residual if line.amount_residual > 0.0 else 0.0,
+            'amount_currency': -line.amount_residual_currency,
+            'account_id': line.account_id.id,
+            'currency_id': line.currency_id.id,
+            'partner_id': line.partner_id.id,
+            'sequence': sequence,
+            }
+
+    def _prepare_exchange_vals_1(self, line, sequence, exchange_line_account):
+        return {
```

Comment:
You probably want to have a single method for both lines. In case you need to search for others business models, you only want to do it once